src/action-creators/pull.ts

```diff
@@ -126,6 +127,26 @@ const pull =
       key => (thoughtsLocal.contextIndex || {})[key].pending,
     )
 
+    // if we are pulling the home context and there are no pending thoughts, but the home parent is marked a pending, it means there are no children and we need to clear the pending status manually
+    // https://github.com/cybersemics/em/issues/1344
+    const state = getState()
+    if (ROOT_ENCODED in contextMap && !hasPending && isPending(state, [HOME_TOKEN])) {
```

Comment:
I think we want to avoid an additional traversal unless it's absolutely necessary. I believe that the reconcile logic ensures that other pending contexts are set to `pending: false` after they load. 

It's checking only ROOT because only ROOT is artificially set to `pending` on initialization. Actually, EM and ABSOLUTE are as well, but they are not visible so the pending status has no visible effect.

Reply:
We want to avoid an additional traversal for sure. But I believe that the `pull` should always produce a valid state irrespective of the other functions (`reconcile` in this case), which may or may not change.

It's more of a judgment call, so I'll suggest you go ahead with whatever seems the best option.

Reply:
Good point. We'll definitely go that direction if we have any more issues with pending. Thanks!